src/fparser/Fortran2003.py

```diff
@@ -6063,7 +6103,27 @@ def match(string):
                 i2 = i2.lstrip()
                 return c, W(i1), M(i2), None
             return c,W(line), None, None
-        if c in ['E','F','G']:
+        if c in ['E']:
+            line = string[1:].lstrip()
+            c2 = line[0].upper()
+            if c2 in ['S', 'N']:
+                line = line[1:].lstrip()
+            else:
+                c2 = ""
+            if line.count('.')==1:
+                i1,i2 = line.split('.',1)
+                i1 = i1.rstrip()
+                i2 = i2.lstrip()
+                return c+c2, W(i1), D(i2), None
+            elif line.count('.')==2:
+                i1,i2,i3 = line.split('.',2)
+                i1 = i1.rstrip()
+                i2 = i2.lstrip()
+                i3 = i3.lstrip()
+                return c+c2, W(i1), D(i2), E(i3)
+            else:
+                return
```

Comment:
True. In fact, looking at this section of the code revealed some bugs. In particular, there was a section that looked for a second decimal point in a format string when in fact it should have been checking for an optional second 'E' character followed by the no. of digits in the exponent. I've corrected this and added tests in efa5513.